### PR TITLE
Add a releaseProfile to generate dependency reduced pom

### PR DESCRIPTION
**Pull request by maobaolong**
#12338

This is a workaround for MSHADE-148, which leads to an infinite loop when building Alluxio.

There are two scenarios.
- Releasing Alluxio
Maven build with `-Prelease`. 
  - If using newer than `3.3.x` version of maven, you also have to use sequential build, Like `mvn clean install -Prelease xxxx`.
  - If using lower version maven before `3.3.x`, you can use parallel build, e.g. `mvn clean install -T 4 -Prelease xxxx`
- Not Releasing Alluxio
If you are not releasing Alluxio, you can build Alluxio like before, please don't add the `-Prelease` into you build command line. e.g. `mvn clean install -T 4 xxxx`

This patch adds a -Prelease. If present, it will set createDependencyReducedPom true. The consequences are:

If you are releasing Alluxio with this profile, you are fine as long as before.
If you are releasing Alluxio without this profile, the `alluxio-shaded-xxx-yyy.pom` will be the same as the pom.xml under the module directory.
If you are not releasing Alluxio but you are using this profile, you may run into #12338
If you are not releasing Alluxio and you did not include this profile, you are fine.
This is all documented in pom.xml and tested locally with maven 3.6.3.

### CODE REVIEW

File: pom.xml
Comment (apc999) [inline pom.xml:1492]: shall we add it here too @Xenorith  `-Prelease-profile`
https://github.com/Alluxio/alluxio/blob/384f944fa6c6986adeca8e598b62feebd7872276/dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go#L99

Comment (Xenorith) [reply pom.xml:1492]: from my understanding but correct me if i'm wrong, `-Prelease-profile` is only necessary when building artifacts to be published to maven. in all other cases, we would not want this profile to be enabled

**Review by apc999 — COMMENTED** [review]
I would just call this `release` instead of `relase-profile`.
otherwise LGTM

@calvinjia  do you want to review this pr too?

File: pom.xml
Comment (Xenorith) [reply pom.xml:1492]: @LuQQiu can you confirm that we do not want to append this profile argument to `generate-tarball.go` because this doesn't get executed when publishing to maven?

Comment (LuQQiu) [reply pom.xml:1492]: Not quite sure if we need it when generating the tarball. But publish to maven doesn't have relationships with generate-tarball.go

Comment (maobaolong) [reply pom.xml]: @LuQQiu Thanks for your reply, as publish to maven doesn't have relationships with generate-tarball.go, so I think we don't need to add `-Prelease` here.

**Comment by maobaolong** [comment]
@apc999 @LuQQiu @Xenorith  Thanks for your review and suggestion, I pushed a new commit to rename the release-profile to release, PTAL.

**Review by Xenorith — APPROVED** [review]
thanks for your efforts to resolve this issue. i verified that building maven without the release profile works. will wait for others to take a look and approve before merging

**Comment by maobaolong** [comment]
@Xenorith Thanks a lot.

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by maobaolong** [comment]
Thank @apc999 for merging this.

**Comment by apc999** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Pull request merged**
